lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,475 @@
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'icon'          => array(
+					'description' => __( 'Icon of block type.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'attributes'    => array(
+					'description'          => __( 'Block attributes.', 'gutenberg' ),
+					'type'                 => 'object',
+					'properties'           => array(),
+					'default'              => array(),
+					'additionalProperties' => array(
+						'type' => 'object',
+					),
+					'context'              => array( 'embed', 'view', 'edit' ),
+					'readonly'             => true,
+				),
+				'supports'      => array(
+					'description' => __( 'Block supports.', 'gutenberg' ),
+					'type'        => 'object',
+					'default'     => array(),
+					'properties'  => array(),
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'category'      => array(
+					'description' => __( 'Block category.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'is_dynamic'    => array(
+					'description' => __( 'Is the block dynamically rendered.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'default'     => false,
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'editor_script' => array(
+					'description' => __( 'Editor script handle.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'script'        => array(
+					'description' => __( 'Public facing script handle.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'editor_style'  => array(
+					'description' => __( 'Editor style handle.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'style'         => array(
+					'description' => __( 'Public facing style handle.', 'gutenberg' ),
+					'type'        => 'string',
+					'default'     => '',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'styles'        => array(
+					'description'          => __( 'Block style variations.', 'gutenberg' ),
+					'type'                 => 'object',
+					'properties'           => array(),
+					'additionalProperties' => array(
+						'type' => 'object',
+					),
+					'default'              => array(),
```

Comment:
I notice that, despite the fact the `type` is assigned as `'object'`, when the default value is returned by the endpoint, it is returned as a JSON array `[]` instead of `{}`. Having dealt with this in the past, I considered if casting it `(object) array()` would be enough, but it appears to make no difference.

Do you know if there's some way to force the default to return as an object?

Reply:
Yeah this is an aggravating issue and there isn't a great way to do it. What can be done is check if `attributes` is empty in `prepare_item_for_response` and set an empty object instead, but I really don't like that because it means that the `attributes` property will sometimes be an array and sometimes be an object.

It is something I'd like to fix at some point, I'll try taking a look at this for 5.5.